crates/shared/src/price_estimation/instrumented.rs

```diff
@@ -141,39 +144,69 @@ mod tests {
         ];
 
         let mut estimator = MockPriceEstimating::new();
-        let expected_query = queries[0].clone();
-        estimator
-            .expect_estimate()
-            .times(1)
-            .withf(move |q| *q == expected_query)
-            .returning(|_| async { Ok(Estimate::default()) }.boxed());
-        let expected_query = queries[1].clone();
-        estimator
-            .expect_estimate()
-            .times(1)
-            .withf(move |q| *q == expected_query)
-            .returning(|_| {
-                async { Err(PriceEstimationError::EstimatorInternal(anyhow!(""))) }.boxed()
-            });
+        let expectations = vec![
+            (0, Ok(Estimate::default())),
+            (1, Err(PriceEstimationError::NoLiquidity)),
+            (
+                0,
+                Err(PriceEstimationError::UnsupportedToken {
+                    token: H160([0; 20]),
+                    reason: "".to_string(),
+                }),
+            ),
+            (
+                1,
+                Err(PriceEstimationError::UnsupportedOrderType("".to_string())),
+            ),
+            (0, Err(PriceEstimationError::RateLimited)),
+            (1, Err(PriceEstimationError::EstimatorInternal(anyhow!("")))),
+            (0, Err(PriceEstimationError::ProtocolInternal(anyhow!("")))),
+        ];
+
+        for (query_index, result) in expectations {
+            let expected_query = queries[query_index].clone();
+            estimator
+                .expect_estimate()
+                .times(1)
+                .withf(move |q| *q == expected_query)
+                .returning(move |_| {
+                    let result = result.clone();
+                    async { result }.boxed()
+                });
+        }
 
         let instrumented = InstrumentedPriceEstimator::new(estimator, "foo".to_string());
 
-        let _ = instrumented.estimate(queries[0].clone()).await;
-        let _ = instrumented.estimate(queries[1].clone()).await;
+        for i in 0..7 {
```

Comment:
That way you can't skip a new test case by accident that was added to the `expectations`.
```suggestion
        for i in 0..expectations.len() {
```